src/snapred/backend/recipe/algorithm/LiteDataCreationAlgo.py

```diff
@@ -4,94 +4,89 @@
 from mantid.api import AlgorithmFactory, PythonAlgorithm, mtd
 from mantid.kernel import Direction
 
+from snapred.backend.dao.RunConfig import RunConfig
+from snapred.backend.recipe.algorithm.LoadGroupingDefinition import LoadGroupingDefinition
 from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
-from snapred.meta.Config import Resource
+from snapred.meta.Config import Config
 
 name = "LiteDataCreationAlgo"
 
 
 class LiteDataCreationAlgo(PythonAlgorithm):
     def PyInit(self):
         self.declareProperty("InputWorkspace", defaultValue="", direction=Direction.Input)
+        self.declareProperty("Run", defaultValue="", direction=Direction.Input)
         self.declareProperty("OutputWorkspace", defaultValue="", direction=Direction.Output)
         self.setRethrows(True)
         self.mantidSnapper = MantidSnapper(self, name)
 
+    def chopIngredients(self, run: RunConfig):
+        self.runNumber = run.runNumber
+        self.IPTS = run.IPTS
+        print(self.IPTS)
+        self.rawDataPath: str = self.IPTS + "nexus/SNAP_{}.nxs.h5".format(self.runNumber)
+        pass
+
     def PyExec(self):
         self.log().notice("Lite Data Creation START!")
-        # define super pixel dimensions
-        superPixel = [8, 8]
 
-        # get output workspace name
+        # load input workspace
+        inputWorkspaceName = self.getProperty("InputWorkspace").value
+
+        # load run number
+        run = RunConfig(**json.loads(self.getProperty("Run").value))
+        self.chopIngredients(run)
+
+        # create outputworkspace
         outputWorkspaceName = self.getProperty("OutputWorkspace").value
 
-        # load input workspace
-        inputWorkspace = self.getProperty("InputWorkspace").value
+        # load file
+        self.mantidSnapper.LoadEventNexus(
+            "Loading Event Nexus for {}...".format(self.rawDataPath),
+            Filename=self.rawDataPath,
+            OutputWorkspace=inputWorkspaceName,
+            NumberOfBins=1,
+            LoadMonitors=False,
+        )
 
-        # clone the workspace
-        self.mantidSnapper.CloneWorkspace(
-            "Cloning input workspace for lite data creation...",
-            InputWorkspace=inputWorkspace,
-            OutputWorkspace=outputWorkspaceName,
+        groupingWorkspaceName = f"{self.runNumber}_lite_grouping_ws"
+
+        # load grouping map
+        self.mantidSnapper.LoadGroupingDefinition(
+            "Loading lite grouping information...",
+            GroupingFilename=str(Config["instrument.lite.map.file"]),
+            InstrumentFilename=str(Config["instrument.native.definition.file"]),
+            OutputWorkspace=groupingWorkspaceName,
```

Comment:
I'm not sure about this comment. The grouping file (the map from full to lite) is a special one for doing the lite conversion. If this were loaded outside of the algo, then we would need an extended recipe that knew to look for it when the request is made to generate a lite version. 

It's true that, at the moment, every run that SNAPRed loads will need to be converted to lite. So it would make sense to persist the lite grouping workspace rather than load it everytime. I just don't know how SNAPRed would know when to load and when to delete if it's done outside of the LiteDataCreation algo??